wasp-ai/src/client/pages/UserPage.jsx

```diff
@@ -0,0 +1,101 @@
+import { useQuery } from "@wasp/queries";
+import getProjectsByUser from "@wasp/queries/getProjectsByUser";
+import { Link } from "@wasp/router";
+import { Color } from "../components/Color";
+import { Header } from "../components/Header";
+import { PiDownloadDuotone } from "react-icons/pi";
+import logout from "@wasp/auth/logout";
+import { FiLogOut } from "react-icons/fi";
+import { format } from "timeago.js";
+import { StatusPill } from "../components/StatusPill";
+import { getColorValue, getStatusName, getStatusText } from "./StatsPage";
+
+export function UserPage({ user }) {
+  const { data: projects, isLoading, error } = useQuery(getProjectsByUser);
+
+  if (isLoading) "Loading Projects...";
+  if (error) "Error loading projects.";
+
+  return (
+    <div className="container">
+      <Header isUserPage />
+      <div className="big-box">
+        <div className="flex items-center justify-between pb-6 pl-1">
+          <p className="text-gray-700 mr-2 whitespace-nowrap">
+            <span className="font-semibold"> {user.username}</span>'s apps
+          </p>
+          <button className="relative group mr-3" onClick={() => logout()}>
+            <FiLogOut className="w-5 h-5 text-slate-600" />
+            <div className="absolute text-center whitespace-nowrap bg-slate-600 text-white text-xs rounded py-1 px-4 bottom-100 left-1/2 transform -translate-x-1/2 translate-y-1 opacity-0 invisible group-hover:opacity-100 group-hover:visible transition-all ease-in-out duration-275">
+              Log out
+            </div>
+          </button>
+        </div>
+        <div className="sm:rounded-lg shadow-md overflow-x-auto ">
+          <table className=" w-full text-sm text-left text-slate-500">
+            <thead className="text-xs text-slate-700 uppercase bg-gray-50">
+              <tr>
+                <th scope="col" className="px-6 py-3">
+                  App Name
+                </th>
+                <th scope="col" className="px-6 py-3">
+                  Status
+                </th>
+                <th scope="col" className="px-6 py-3">
+                  Created At
+                </th>
+                <th scope="col" className="px-6 py-3">
+                  Creativity lvl
+                </th>
+                <th scope="col" className="px-6 py-3"></th>
+              </tr>
+            </thead>
+            <tbody>
+              {projects?.length &&
```

Comment:
<img width="674" alt="Screenshot 2023-10-30 at 11 44 58" src="https://github.com/wasp-lang/wasp/assets/2223680/7755d1ad-3c84-435c-871e-17796f714ea6">

To avoid this always use `.length > 0`

Reply:
Interesting, I would expect 0 to be false! Why would he need to add > 0?

Reply:
yeah I assumed the same, but seems this is how react behaves...

Reply:
Aha, ok so what happens is following:

You have `project?.length && projects.map(...)`.

So, if the first part of `&&` is "falsy", e.g. `undefined` or `0`, then that is the value of the whole expression, and what React gets to render is `undefined` or `0`.

On the other hand, if it is "truthy", e.g. `1` or something, then it will render that `project.map(...)`.

So what happens when React gets `undefined` or `0` to render? Well, if it gets `undefined`, I guess it doesn't render anything, but if it gets `0`, it renders `0`! So that is the issue here.

What @infomiho wrote will help because it will ensure the first part of `&&` is boolean: True or False (since `>` is boolean operator). And React, when given `False`, doesn't render anything, so that is good.

You could also do `!!(project?.length)`.